### PR TITLE
Add GitHub action for setting up the Go toolchain

### PR DESCRIPTION
**Pull request by viktorvoltaire**
With this, we can use something like

```
    - name: Set up Go Tooling
      uses: einride/sage
        cacheKey: build
```

### CODE REVIEW

**Review by Edholm — APPROVED** [review]
woopwoop!

**Comment by Edholm** [comment]
This allows for much simpler (and faster) GitHub action workflows :man_dancing:

**Comment by odsod** [comment]
Do we need to work with nampespacing a little bit here?

**Comment by odsod** [comment]
Like sage/actions/setup-go?

**Comment by Edholm** [comment]
> - name: Set up Go Tooling
>       uses: einride/sage
>         cacheKey: build

Do you mean this @odsod ?

It isn't immediately apparent which action is being invoked by this. Dunno if we even can improve it? :thinking:

**Comment by Edholm** [comment]
Musn't the action.yaml file be put under .github/actions/<name>/action.yml ? :thinking:

**Comment by viktorvoltaire** [comment]
> Like sage/actions/setup-go?

Sure, I was a bit uncertain first what was allowed. But seems like any path would work. Agreed that `uses: einride/sage/actions/setup-go` is much better

**Comment by viktorvoltaire** [comment]
> Musn't the action.yaml file be put under .github/actions//action.yml ? thinking

the `uses' attribute must be a path, a Docker image, or owner/repo@ref

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        restore-keys: |
+          ${{ runner.os }}-go-${{ inputs.cacheKey }}-
+
+    - name: Cache sage folders
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./.sage/tools
+          ./.sage/bin
+        key: ${{ runner.os }}-sage-${{ inputs.cacheKey }}-${{ hashFiles('./.mage/go.sum') }}
```
Comment (stenehall) [inline action.yml:35]: should this be `./.sage/`?

**Comment by viktorvoltaire** [comment]
I can try and make a prerelease of this branch, and try different paths @odsod @Edholm and we can see if it works

**Comment by viktorvoltaire** [comment]
@odsod @Edholm this worked! Should we go with that?

```
      - name: Setup Go toolchain
        uses: einride/sage/actions/setup-go@v0.82.0-prerelease
        with:
          cacheKey: build
```

**Comment by odsod** [comment]
@viktorvoltaire Nice - do we need to manually specify a cache key by default?

**Comment by viktorvoltaire** [comment]
> @viktorvoltaire Nice - do we need to manually specify a cache key by default?

No its optional
```
    description: Custom cache key used
    required: false
    default: ci
```

**Comment by viktorvoltaire** [comment]
For this repo, i guess we could skip it given that we just run make

File: action.yml (hunk 1 above)
Comment (Edholm) [reply action.yml]: We only want to cache generated files. We don't want to cache the go files themselves :)

**Comment by Edholm** [comment]
> @viktorvoltaire Nice - do we need to manually specify a cache key by default?

The reason I added it originally is that if you have multiple jobs having the same action, they end up having the same cache key which can conflict. 

I.e. you can end up in situation where the cache for a job is incomplete because it was created by a different job.

**Review by Edholm — APPROVED** [review]
woop woop :+1:

**Review by Edholm — APPROVED** [review]

**Pull request merged**